tests/test/codegen/test_machine_code_generator.cpp

```diff
@@ -219,6 +217,9 @@ TEST_CASE_METHOD(CodegenTestFixture,
     // Run the codegen
     gen.codegenForSharedObject(localSharedObjFile);
 
+    // Check the locally cached path matches the expected one
+    REQUIRE(loader.getSharedObjectObjectFile(localSharedObjFile) == objFile);
```

Comment:
With the new `path.append` behaviour in `std::filesystem` this check would have failed.

This is, before:
`path.append("/tmp/obj", "/usr/local/faasm/foo/bar.o"); // = "/usr/local/faasm/foo/bar.o")`
Now:
`"/tmp/obj" += "/usr/local/faasm/foo/bar.o"; // = "/tmp/obj/usr/local/faasm/foo/bar.o"`